pkgs/applications/networking/cluster/terraform-providers/update-provider

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p bash coreutils curl jq
+# shellcheck shell=bash
+#
+# Update a terraform provider to the latest version advertised at
+# the provider source address.
+set -euo pipefail
+
+# TODO: document usage
+provider_name="$1"
```

Comment:
Do you see an easy way to list and/or update all of the providers from the registry?

Reply:
I'm inclined to provide two additional scripts, one to add a provider to `providers.json` from it's registry URL (as opposed to exhaustively add all providers). And one script to update all known providers.

```bash
# update-all-providers
jq 'keys[]' providers.json | xargs ./update-provider
```

I should be able to also re-use much of `update-provider` to create `add-provider registry.terraform.io/some/provider`

Any thoughts?

I will absolutely look at using `update-provider` with the `updateScript` framework :heart:

Reply:
Everything you said sounds good to me!